### PR TITLE
Issue 4093 - fix compiler warnings and update doxygen

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:  

Update the doxy file (doxygen), fix compiler warnings (x86_64, arm, and s390x), and update Rust cargo file.

relates: https://github.com/389ds/389-ds-base/issues/4093

### CODE REVIEW

File: ldap/servers/plugins/acctpolicy/acct_util.c [hunk 1]
```diff
@@ -27,7 +27,7 @@ Hewlett-Packard Development Company, L.P.
 #include "slapi-plugin.h"
 
 /* Globals */
-static void *plugin_id = NULL;
+static void *global_plugin_id = NULL;
```
Comment (Firstyear) [inline ldap/servers/plugins/acctpolicy/acct_util.c:32]: Worth checking with globals we have ODR turned on.

File: ldap/servers/plugins/acl/acl.c [hunk 2]
```diff
@@ -1655,15 +1655,15 @@ acl_modified(Slapi_PBlock *pb, int optype, Slapi_DN *e_sdn, void *change)
...
-                    rv = slapi_entry_attr_find(e, "uniqueMember", &a);
-                    if (rv != 0)
+                    int rc;
+                    rc = slapi_entry_attr_find(e, "uniqueMember", &a);
+                    if (rc != 0)
                         break;
-                    rv = slapi_entry_attr_find(e, "Member", &a);
-                    if (rv != 0)
+                    rc = slapi_entry_attr_find(e, "Member", &a);
+                    if (rc != 0)
```
Comment (Firstyear) [inline ldap/servers/plugins/acl/acl.c]: {} since were touching it anyway.

**Review by Firstyear — APPROVED** [review]
Some small nits, but otherwise looks really good. Since it's a large change, if @progier389 has some spare time could he look at it too?

File: ldap/servers/slapd/auth.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /** BEGIN COPYRIGHT BLOCK
  * Copyright (C) 2001 Sun Microsystems, Inc. Used by permission.
- * Copyright (C) 2005 Red Hat, Inc.
+ * Copyright (C) 2021 Red Hat, Inc.
```
Comment (progier389) [inline ldap/servers/slapd/auth.h:3]: Why changing copyright in this file and not in the others ?

File: ldap/servers/slapd/tools/dbscan.c [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /** BEGIN COPYRIGHT BLOCK
- * Copyright (C) 2005 Red Hat, Inc.
+ * Copyright (C) 2021 Red Hat, Inc.
```
Comment (progier389) [inline ldap/servers/slapd/tools/dbscan.c]: Should we change copyright on a file that is not modified otherwise ?

**Review by progier389 — APPROVED** [review]
Thats Ok for me once some points about copyright are answered

File: ldap/servers/slapd/tools/dbscan.c (hunk 4 above)
Comment (mreynolds389) [reply ldap/servers/slapd/tools/dbscan.c]: So what happened here is that I had made changes to these files, but I undid them for some reason (probably from when I attempted to fix all the discarded qualifier warnings and then aborted that effort).  

I will undo these copyright changes...

File: ldap/servers/slapd/auth.h (hunk 3 above)
Comment (mreynolds389) [reply ldap/servers/slapd/auth.h:3]: Yeah, it was hard to keep track of which ones I did and did not do.  This patch was a bit of nightmare and I lost track of keeping the copyrights consistently updated.  I'll fix this.

File: ldap/servers/slapd/tools/dbscan.c (hunk 4 above)
Comment (mreynolds389) [reply ldap/servers/slapd/tools/dbscan.c]: We should not, this is all fixed now

File: ldap/servers/plugins/acl/acl.c (hunk 2 above)
Comment (mreynolds389) [reply ldap/servers/plugins/acl/acl.c]: done

File: ldap/servers/plugins/acctpolicy/acct_util.c (hunk 1 above)
Comment (mreynolds389) [reply ldap/servers/plugins/acctpolicy/acct_util.c:32]: No ODR errors

**Comment by mreynolds389** [comment]
All changes applied, please review...

**Comment by mreynolds389** [comment]
> All changes applied, please review...

We have a problem where I replaced strncpy with strcpy because strcpy adds a NUL terminator and in some cases that breaks things like ACI parsing.  Looks like I need to use memcpy instead.  Still investigating...

**Comment by mreynolds389** [comment]
Changes made, this is ready for review again...

**Review by Firstyear — APPROVED** [review]
Looks good to me!

**Review by droideck — APPROVED** [review]
Just went through the changes, looks good to my knowledge!

**Pull request merged**